examples/Billing/AddBillingSetup.cs

```diff
@@ -0,0 +1,205 @@
+            {
+                // If a payments account id has been provided, set PaymentsAccount to its resource
+                // name. You can list available payment accounts via the PaymentAccountService's
+                // ListPaymentsAccounts method.
+                billingSetup.PaymentsAccount =
+                    ResourceNames.PaymentsAccount(customerId, paymentsAccountId);
+            }
+            else
+            {
+                // Otherwise, create a new payments account by setting the PaymentsAccountInfo
+                // field. See https://support.google.com/google-ads/answer/7268503 for information
+                // about payments profiles.
+                if (paymentsProfileId == null)
+                {
+                    throw new Exception(
+                        "No paymentsAccountId or paymentsProfileId provided.");
+                }
+
+                billingSetup.PaymentsAccountInfo = new BillingSetup.Types.PaymentsAccountInfo()
+                {
+                    PaymentsAccountName = "Payments Account #" + ExampleUtilities.GetRandomString(),
+                    PaymentsProfileId = paymentsProfileId
+                };
+            }
+
+            return billingSetup;
+        }
+
+        /// <summary>
+        /// Sets the starting date time for the new billing setup. Queries the customer's account
+        /// to see if there are any approved billing setups. If there are any, the new billing setup
+        /// may have a starting date time of any day in the future. If not, the billing setup must
+        /// be set to start immediately.
+        /// </summary>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        /// <param name="googleAdsService">The Google Ads service client.</param>
+        /// <param name="billingSetup">The instance of BillingSetup whose starting date time will
+        ///     be set.</param>
+        private void SetBillingSetupStartDateTime(long customerId, GoogleAdsServiceClient
+            googleAdsService, BillingSetup billingSetup)
+        {
+            // Query to see if there are any existing approved billing setups. See
+            // GetBillingSetup.cs for a more detailed example of requesting billing setup
+            // information.
+            string query = @"
+                    SELECT billing_setup.id, billing_setup.status
+                    FROM billing_setup
+                    WHERE billing_setup.status = 'APPROVED' LIMIT 1";
+
+            PagedEnumerable<SearchGoogleAdsResponse, GoogleAdsRow> searchResponse =
+                googleAdsService.Search(customerId.ToString(), query);
+
+            if (searchResponse.Any())
+            {
+                // If there is an existing approved billing setup, then you can specify any future
+                // start date for the billing setup.
+                billingSetup.StartDateTime = DateTime.Today.AddDays(1).ToString("yyyyMMdd");
```

Comment:
You may want to request the end date of this billing setup to demonstrate how to make the new one start right after the existing one? Adding 1 day can impact the existing Billing Setup, we should try to avoid this if possible.

Reply:
Fixed to query for latest ending time and to start the new billing setup one day later.